apps/entity-service/src/entities/processors/nursery-report.processor.ts

```diff
@@ -0,0 +1,183 @@
+      } else if (query.sort.field === "organisationName") {
+        builder.order(["nursery", "project", "organisation", "name", query.sort.direction ?? "ASC"]);
+      } else if (query.sort.field === "projectName") {
+        builder.order(["nursery", "project", "name", query.sort.direction ?? "ASC"]);
+      } else if (query.sort.field !== "id") {
+        throw new BadRequestException(`Invalid sort field: ${query.sort.field}`);
+      }
+    }
+
+    const frameworkPermissions = permissions
+      ?.filter(name => name.startsWith("framework-"))
+      .map(name => name.substring("framework-".length) as FrameworkKey);
+    if (frameworkPermissions?.length > 0) {
+      builder.where({ frameworkKey: { [Op.in]: frameworkPermissions } });
+    } else if (permissions?.includes("manage-own")) {
+      builder.where({ "$nursery.project.id$": { [Op.in]: ProjectUser.userProjectsSubquery(userId) } });
+    } else if (permissions?.includes("projects-manage")) {
+      builder.where({ "$nursery.project.id$": { [Op.in]: ProjectUser.projectsManageSubquery(userId) } });
+    }
+
+    const associationFieldMap = {
+      nurseryUuid: "$nursery.uuid$",
+      organisationUuid: "$nursery.project.organisation.uuid$",
+      country: "$nursery.project.country$",
+      projectUuid: "$nursery.project.uuid$"
+    };
+
+    for (const term of [
+      "status",
+      "updateRequestStatus",
+      "frameworkKey",
+      "nurseryUuid",
+      "organisationUuid",
+      "country",
+      "projectUuid"
+    ]) {
+      if (query[term] != null) {
+        const field = associationFieldMap[term] || term;
+        builder.where({ [field]: query[term] });
+      }
+    }
+
+    if (query.search != null) {
+      builder.where({
+        [Op.or]: [
+          { "$nursery.project.name$": { [Op.like]: `%${query.search}%` } },
+          { "$nursery.project.organisation.name$": { [Op.like]: `%${query.search}%` } }
+        ]
+      });
+    }
+
+    if (query.nurseryUuid != null) {
+      const nursery = await Nursery.findOne({ where: { uuid: query.nurseryUuid }, attributes: ["id"] });
+      if (nursery == null) {
+        throw new BadRequestException(`Nursery with uuid ${query.nurseryUuid} not found`);
+      }
+      builder.where({ nurseryId: nursery.id });
+    }
+
+    return { models: await builder.execute(), paginationTotal: await builder.paginationTotal() };
+  }
+
+  async addFullDto(document: DocumentBuilder, nurseryReport: NurseryReport): Promise<void> {
+    const nurseryReportId = nurseryReport.id;
+    const reportTitle = await this.getReportTitle(nurseryReport);
+    const projectReportTitle = await this.getProjectReportTitle(nurseryReport);
+    const readableCompletionStatus = await this.getReadableCompletionStatus(nurseryReport.completion);
+    const createdByUser = await User.findOne({ where: { id: nurseryReport?.createdBy } });
+    const approvedByUser = await User.findOne({ where: { id: nurseryReport?.approvedBy } });
```

Comment:
These should be getting included in the initial fetch with a join; no reason to pull them in separate queries after the fact.